src/reanimated2/animation/repeat.ts

```diff
@@ -100,7 +108,14 @@ export const withRepeat = function <T extends AnimationObject>(
         now: Timestamp,
         previousAnimation: Animation<any> | null
       ): void {
-        animation.startValue = value;
+        // Detect re-render
+        const hasBeenInterrupted =
+          value !== initialAnimationBounds.start &&
+          value !== initialAnimationBounds.end;
+        const startingValue = hasBeenInterrupted
+          ? initialAnimationBounds.start ?? value
+          : value;
```

Comment:
I think that this is easier to read:

```suggestion
        const startingValue = (hasBeenInterrupted && initialAnimationBounds.start)
          ? initialAnimationBounds.start
          : value;
```